BeeSwift/GoalViewController.swift

```diff
@@ -368,12 +371,14 @@ class GoalViewController: UIViewController,  UIScrollViewDelegate, DatapointTabl
         var components = DateComponents()
         components.day = Int(self.dateStepper.value)
 
-        let newDate = (calendar as NSCalendar?)?.date(byAdding: components, to: Date(), options: [])
+        let now = Date()
+        guard let newDate = calendar.date(byAdding: components, to: now) else { return }
```

Comment:
Nit: This seems like the return here should never happen. I like logging a warning in these cases.